resources.go

```diff
@@ -191,17 +191,7 @@ func preConfigureCallback(vars resource.PropertyMap, c *terraform.ResourceConfig
 	}
 	config.CredsFilename = credsPath
 
-	// TODO[pulumi/pulumi-terraform#48] We should also be setting `config.AssumeRole*` here, but we are currently
-	// blocked on not being able to read out list-valued provider config.
-
-	creds, err := awsbase.GetCredentials(config)
-	if err != nil {
-		return errors.New("unable to discover AWS AccessKeyID and/or SecretAccessKey " +
-			"- see https://pulumi.io/install/aws.html for details on configuration")
-	}
-
-	_, err = creds.Get()
-	if err != nil {
+	if _, err := awsbase.GetCredentials(config); err != nil {
```

Comment:
I'm not sure I follow why this is the right fix.

All of the code here is intended to catch and report failures to authenticate prior to the Terraform Provider `Configure` call reporting back failures.  But the goal is to do the same checks the Terraform Provider will do.

Per https://github.com/hashicorp/aws-sdk-go-base/blob/8fac28668e59a2e5afa6b2d034614c648ddc418e/session.go#L31, it appears the call to `creds.Get()` will be made there.  

Is the problem instead that `Profile:   stringValue(vars, "profile"),` isn't correctly reading the profile here?

Reply:
The reason using `AWS_PROFILE` was failing is because we weren't reading the profile correctly before - whether through a profile configured in `~/.aws/config` or `~/.aws/credentials`.

I don't think removing the call to `Get` on the returned `Credentials` struct is _strictly_ necessary to fix the bug, but I also don't think it buys us anything to do it. `GetCredentials` itself calls `Get` here (after constructing a valid credentials chain):

https://github.com/hashicorp/aws-sdk-go-base/blob/master/awsauth.go#L239

It also tests that a role can be assumed, if appropriate:

https://github.com/hashicorp/aws-sdk-go-base/blob/master/awsauth.go#L283

If we hit either of these cases, the call to `GetCredentials` would have failed before _anyway_, so `Get` would not be called. If it worked the first time (in `GetCredentials`), calling it a second time should not yield different behaviour without configuration changes.

It's worth noting that the `aws-sdk-go-base` library is newer than the Pulumi AWS provider, and that the checks done by Terraform have changed as a result of using this library.

Finally, the error message we produce in response to _any_ error from configuration complains about `AccessKeyID` and `SecretAccessKey`. In future, we should change the error to reflect the actual source of the problem. We _could_ duplicate the entire logic of `GetCredentials` in the pre-configure callback to give better error messages here, but it like we'd be better off doing string detection on the error values to avoid doing so.

Reply:
>  don't think removing the call to Get on the returned Credentials struct is strictly necessary to fix the bug, but I also don't think it buys us anything to do it. 

Got it - makes sense.

> Finally, the error message we produce in response to any error from configuration complains about AccessKeyID and SecretAccessKey. In future, we should change the error to reflect the actual source of the problem. We could duplicate the entire logic of GetCredentials in the pre-configure callback to give better error messages here, but it like we'd be better off doing string detection on the error values to avoid doing so.

Totally agreed.